### PR TITLE
Arktos Pod Creation Integration

### PR DESCRIPTION
**Pull request by clu2xlu**

### CODE REVIEW

File: etc/crds/nets.crd.yaml [hunk 1]
```diff
@@ -32,8 +32,8 @@ spec:
       storage: true
   names:
     kind: Net
```
Comment (phudtran) [inline etc/crds/nets.crd.yaml:34]: This should also be changed to subnet

Comment (phudtran) [inline etc/crds/nets.crd.yaml]: Can we change this to subnet to be consistent with the terminology?

File: mizar/dp/mizar/operators/endpoints/endpoints_operator.py [hunk 2]
```diff
@@ -239,16 +258,13 @@ def produce_simple_endpoint_interface(self, ep):
...
-        # TODO (cathy): mark the pod network as ready! Shall it be here or in
-        # the pod builtin wf.
         logger.info("Produced {}".format(interfaces))
 
     def create_simple_endpoints(self, interfaces, spec):
         """
         Create a simple endpoint object (calling the API operator)
         """
-        for interface in interfaces.interfaces:
+        for interface, net_info in zip(interfaces.interfaces, spec['interfaces']):
```
Comment (phudtran) [inline mizar/dp/mizar/operators/endpoints/endpoints_operator.py:267]: Instead of looking in spec['interfaces'] again here, can you just add  the needed information to the interface message/object?

See here.
https://github.com/futurewei-cloud/mizar/blob/dev-next/mizar/dp/mizar/operators/endpoints/endpoints_operator.py#L313

File: mizar/obj/net.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def get_name(self):
         return self.name
 
     def get_plural(self):
-        return "nets"
+        return "subnets"
 
     def get_kind(self):
         return "Net"
```
Comment (phudtran) [inline mizar/obj/net.py]: This should be subnet also to be consistent.

File: mizar/obj/tests/test_net2.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+# substantial portions of the Software.THE SOFTWARE IS PROVIDED "AS IS",
+# WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+# TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+# FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
+# THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+apiVersion: "mizar.com/v1"
+kind: Net
```
Comment (phudtran) [inline mizar/obj/tests/test_net2.yaml]: subnet

File: etc/crds/nets.crd.yaml [hunk 5]
```diff
@@ -32,8 +32,8 @@ spec:
...
-    plural: nets
-    singular: net
+    plural: subnets
+    singular: subnet
     shortNames:
       - net
       - nets
```
Comment (phudtran) [inline etc/crds/nets.crd.yaml]: ditto

File: mizar/dp/mizar/operators/endpoints/endpoints_operator.py [hunk 6]
```diff
@@ -258,12 +274,14 @@ def create_simple_endpoints(self, interfaces, spec):
 
             ep.set_vni(spec['vni'])
             ep.set_vpc(spec['vpc'])
-            ep.set_net(spec['net'])
+            ep.set_net(net_info.get('subnet', spec['net']))
```
Comment (phudtran) [inline mizar/dp/mizar/operators/endpoints/endpoints_operator.py]: With changing the kind of the net crd to "subnet" we will need to look into spec["subnet"] here.

**Review by phudtran — CHANGES_REQUESTED** [review]
Overall looks good. Some very minor comments.
Thanks Cathy!

File: mizar/dp/mizar/operators/endpoints/endpoints_operator.py (hunk 2 above)
Comment (clu2xlu) [reply mizar/dp/mizar/operators/endpoints/endpoints_operator.py:267]: This not going to work tho, https://github.com/clu2xlu/mizar/blob/cni-delete-new/mizar/dp/mizar/workflows/builtins/pods/create.py#L60
This is the interfaces defined in the yaml file when calling kubectl create.

File: cli/mizarapi.py [hunk 7]
```diff
@@ -36,13 +36,13 @@ def create_net(self, name, ip, prefix, vpc, bouncers=1, vni=None):
             "vpc": vpc,
             "bouncers": bouncers
         }
-        self.create_obj(name, "Net", spec, "nets")
```
Comment (XiaoningDing) [inline cli/mizarapi.py:39]: It's better to split the renaming part as a separate PR.   It will make review easier and commits more portable.

Comment (clu2xlu) [reply cli/mizarapi.py:39]: Pod create move to https://github.com/futurewei-cloud/mizar/pull/224
I only created PR for pod create (not change naming to net to subnet) is because part of the net to subnet change is based off of pod create PR. I am hoping to merge my pod create PR first before I create PR for net to subnet change

Comment (XiaoningDing) [reply cli/mizarapi.py:39]: Is this PR still valid? I see there are lots of duplication between this PR and your separated PR.

Comment (clu2xlu) [reply cli/mizarapi.py:39]: NO, will close this one
